src/main/java/carpet/script/language/DataStructures.java

```diff
@@ -341,7 +341,7 @@ else if (lv.size() == 2)
         expression.addUnaryFunction("encode_json", v -> StringValue.of(v.toJson().toString()));
         expression.addUnaryFunction("decode_json", v -> {
             try {
-                return new ScarpetJsonDeserializer().deserialize(new JsonParser().parse(v.getString()), null, null);
+                return new ScarpetJsonDeserializer().deserialize(JsonParser.parseString(v.getString()), null, null);
```

Comment:
This is wrong, the deserializer shouldn't be called by just nulling out parameters that may be required later. Not only that, this warning is already fixed in another PR.

Reply:
ok, i revert the latest commit.